examples/file-uploader/js/list.js

```diff
@@ -0,0 +1,87 @@
+const Type = require('union-type');
+const T = require('ramda/src/T')
+    , adjust = require('ramda/src/adjust')
+    , append = require('ramda/src/append')
+    , curry  = require('ramda/src/curry')
+;
+const h = require('snabbdom/h');
+
+const upload = require('./upload');
+const uploader = require('./uploader');
+
+const sync = (s) => [s, []];
+
+// note: prefer to check if iterable, 
+// but FileList.prototype doesn't seem to have Symbol.iterator cross-browser?
+const isFileList = (x) => !(undefined === x.length) 
+
+// action
+
+const Action = Type({
+  Create:      [Function, isFileList],
+  Result:      [Number, uploader.Result]
+});
+
+const update = Action.caseOn({
+
+  Create: (up,files,model) => {
+    const idx = nextIndex(model);
+    const task = up(files);
+    const taskAction = Action.Result(idx);
+    const newState = append( upload.init(files), model);
+    return [newState, [task.map(taskAction)]];
+  },
+  
+  Result: (i,result,model) => {
+    const finish = (type) => () => {
+      return adjust(upload.update(upload.Action[type]()), i, model);
+    };
+    return sync(
+      uploader.Result.case({
+        OK:       finish('Uploaded'),
+        NotFound: finish('Error'),
+        Error:    finish('Error'),
+        Abort:    finish('Abort'), 
+        Progress: (abort,p) => {
+          return adjust(upload.update(upload.Action.Progress(abort,p)), i, model);
+        }
+      }, result)
+    );
+  }
+
+});
+
+
+// model
+
+const init = () => []
+const nextIndex = (model) => model.length;
+
+// view
+
+const view = (model) => {
+
+  const style = {'list-style': 'none',
+                 '-webkit-margin-before': 0,
+                 '-webkit-margin-after': 0,
+                 '-webkit-padding-start': 0
+                };
+
+  const listItemView = (item, i) => {
+    const substyle = { };
```

Comment:
Originally I intended to parameterize styles, didn't quite finish, but I think it makes the example harder to read so will probably take this out